core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostEstimator.scala

```diff
@@ -36,7 +39,15 @@ import scala.util.{Failure, Success, Try}
   * A cost estimator should estimate a cost of running a region under the given resource constraints as units.
   */
 trait CostEstimator {
-  def estimate(region: Region, resourceUnits: Int): Double
+
+  /**
+    * Uses the given resource units to allocate resources to the region, and determine a cost based on the allocation.
+    *
+    * Note currently the ResourceAllocator is not cost-based and thus we use a cost model that does not rely on the
+    * allocator, i.e., the cost estimation process is external to the ResourceAllocator.
+    * @return An updated region with allocated resources and an estimated cost of this region.
```

Comment:
for the return, I suggest we only return resource config and a cost. the caller should be responsible to evaluate the cost and decide the resource config to attach to the region.